lib/rules/indent.js

```diff
@@ -807,6 +889,84 @@ module.exports = {
             });
         }
 
+        /**
+         * Iterate through the dependent tokens until we reach the last token
+         * dependency or the last token dependency that is within the unknown
+         * node. If the last token dependency is not the first token of the line
+         * make the last token dependcy depend on the first token of the line.
+         * This ensures the offset will be set correctly since only the first
+         * token of each line is ignored and not all tokens on a line.
+         * @param {Token} dependency First
+         * @param {number} startLineOfNode First line number of unknown node
+         * @returns {void}
+         */
+        function ignoreUnknownNodeLineWithDependency(dependency, startLineOfNode) {
+            let currDependency = dependency,
+                lastDependency = dependency;
+
+            while (
+                (currDependency = offsets.getFirstDependency(currDependency)) &&
+                currDependency.loc.start.line > startLineOfNode
```

Comment:
The solution you posted works for the code snippet I posted, it's a great idea. There is one other case that this does not account for. When an unknown node is within a known node. There is a situation where a token can have its offset depend on tokens outside of the unknown node. This will cause the `getDesiredIndent()` to keep traversing the depndencies and not stop on the ignored line.

For example:
```ts
1 function foo() {
2   function bar() {
3     abstract class X {
4       public baz() {
5         qux();
6       }
7     }
8   }
9 }
```

In this case Line 5 depends on Line 4 tokens `{` and `(`. The `(` token depends on Line 2 tokens `{` and `function`. None of the tokens are dependent on any tokens in Line 3. 

We need a way to break the token dependency chain when it continues outside of an unknown node. One solution is to check if the dependency of a line, inside an unknown node, lies outside the unknown node, then remove the dependency. It is simple to check if dependencies lies outside the unknown node.

The problem is how to determine which token is used a dependency in the line. I can just iterate through all the tokens on the line and set their dependencies to null but I feel like this could be improved. Maybe if I go down a line and traverse the dependencies but that feels like the old solution again.

Any ideas on how to handle this case or is removing the dependency of every token on the line sufficient?